iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/TopologyModificationUtils.java

```diff
@@ -161,6 +207,68 @@ private static <L extends LoadingLimits, A extends LoadingLimitsAdder<L, A>> voi
         adder.add();
     }
 
+    static LoadingLimitsBag calcMinLoadingLimitsBag(List<LoadingLimits> loadingLimits) {
+        LoadingLimitsBag limit = new LoadingLimitsBag();
+        List<TemporaryLimitsBag> temporaryLimitsBags = new ArrayList<>();
+        Map<Integer, TemporaryLimitsBag> temporaryLimitsByAcceptableDuration = new HashMap<>();
+        double minPermanentLimitValue = Double.MAX_VALUE;
+
+        for (LoadingLimits loadingLimit : loadingLimits) {
+            // find the min permanent limit value
+            if (loadingLimit.getPermanentLimit() < minPermanentLimitValue) {
+                minPermanentLimitValue = loadingLimit.getPermanentLimit();
+            }
+
+            // find the min temporary limit value for each acceptableDuration
+            Collection<LoadingLimits.TemporaryLimit> temporaryLimits = loadingLimit.getTemporaryLimits();
+            for (LoadingLimits.TemporaryLimit temporaryLimit : temporaryLimits) {
+                Integer acceptableDuration = temporaryLimit.getAcceptableDuration();
+                if (!temporaryLimitsByAcceptableDuration.containsKey(acceptableDuration)) {
+                    temporaryLimitsByAcceptableDuration.put(acceptableDuration, new TemporaryLimitsBag(temporaryLimit));
+                }
+                if (temporaryLimit.getValue() < temporaryLimitsByAcceptableDuration.get(acceptableDuration).getValue()) {
```

Comment:
An 'else' in between would help understanding both execution flow are exclusive.
And even reversing the order would remove the negation in the test.
And for sake of minimality, I would have searched only once by storing result of the get and test if against null. But I think it can not be sold as really performance improving, as loadingLimits may rarely be numerous ;-)